pkg/cluster/internal/create/kubeadm-init.go

```diff
@@ -103,11 +104,28 @@ func runKubeadmInit(ec *execContext, configNode *NodeReplica) error {
 
 	// install the CNI network plugin
 	// TODO(bentheelder): support other overlay networks
-	if err := node.Command(
-		"/bin/sh", "-c",
-		`kubectl apply --kubeconfig=/etc/kubernetes/admin.conf -f "https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version --kubeconfig=/etc/kubernetes/admin.conf | base64 | tr -d '\n')"`,
-	).Run(); err != nil {
-		return errors.Wrap(err, "failed to apply overlay network")
+	// first probe for a pre-installed manifest
+	haveDefaultCNIManifest := true
+	if err := node.Command("test", "-f", "/kind/manifests/default-cni.yaml").Run(); err != nil {
+		haveDefaultCNIManifest = false
+	}
+	if haveDefaultCNIManifest {
+		// we found the default manifest, install that
+		// the images should already be loaded along with kubernetes
+		if err := node.Command(
+			"kubectl", "create", "--kubeconfig=/etc/kubernetes/admin.conf",
```

Comment:
bellow we are using the `"/bin/sh", "-c", ...` patter for kubectl.
this seems better.

Reply:
we still pass the kubeconfig via the flag? this is the most explicit way to pass kubeconfig, I don't think using bash is better, we just need to easily exactly mimic the command they specify in their docs

Reply:
I would've strongly preferred to _not_ invoke bash, but the upstream preferred installation from a non-fixed manifest is:

> `kubectl apply -f "https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version | base64 | tr -d '\n')"`

https://www.weave.works/docs/net/latest/kubernetes/kube-addon/#-installation